### PR TITLE
Add formatting scripts

### PR DESCRIPTION
**Pull request by jmmaljaars**
Adding the scripts for `clang` formatting the source code to make it google style compliant. As for now, it's a merge request, awaiting a green light for a final merge of the `development` branch (once any remaining and relevant pull requests have been merged into the `development` branch) and subsequently running:

```
./scripts/run-clang-format.sh -i "*.h" -i "*.cc"
```
(or just `./scripts/run-clang-format.sh`)

from the source root directory.

### CODE REVIEW

File: scripts/clang-format-check.sh [hunk 1]
```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
```
Comment (mnijhuis-tos) [inline scripts/clang-format-check.sh:1]: Add copyright header to this file.

File: scripts/clang-format-check.sh [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+else
+    EX_DIR+=$(join_by " " "${exclude_directories[@]}")
+fi
+
+# Run clang-format from run-clang-format.sh
+SCRIPT_PATH=$(dirname "$0")
+$SCRIPT_PATH/run-clang-format.sh $EX_DIR $INC_FILE
+
+# Can't use tee because it hides the exit code
+if git diff --patch --exit-code > $PATCH_NAME; then
```
Comment (mnijhuis-tos) [inline scripts/clang-format-check.sh:76]: When we are removing $PATCH_NAME anyway, why not use /dev/null, or a quiet diff?

File: scripts/run-clang-format.sh [hunk 3]
```diff
@@ -0,0 +1,72 @@
+
```
Comment (mnijhuis-tos) [inline scripts/run-clang-format.sh:1]: Add copyright header to this file.

File: scripts/run-clang-format.sh [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+#!/bin/bash
```
Comment (mnijhuis-tos) [inline scripts/run-clang-format.sh:2]: Remove the first empty line. I thought #!/bin/bash should be on the first line.

File: scripts/clang-format-check.sh (hunk 2 above)
Comment (jmmaljaars) [reply scripts/clang-format-check.sh:76]: Good catch! Will change to `--quiet` mode

**Comment by jmmaljaars** [comment]
Could you do a final check and approve @mnijhuis-tos

**Pull request merged**